src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java

```diff
@@ -0,0 +1,217 @@
+import com.browserstack.automate.AutomateClient;
+import com.browserstack.automate.ci.common.Tools;
+import com.browserstack.automate.ci.common.enums.ProjectType;
+import com.browserstack.automate.exception.BuildNotFound;
+import com.browserstack.automate.model.Build;
+import com.browserstack.automate.model.Session;
+import com.browserstack.client.BrowserStackClient;
+import com.browserstack.client.exception.BrowserStackException;
+import hudson.model.Run;
+import org.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.browserstack.automate.ci.common.logger.PluginLogger.log;
+
+public class BrowserStackReportForBuild extends AbstractBrowserStackReportForBuild {
+    private static final int RESULT_META_MAX_SIZE = 5;
+    private final String buildName;
+    private final List<Session> browserStackSessions;
+    private final List<JSONObject> result;
+    private final List<JSONObject> resultMeta;
+    private final JSONObject resultAggregation;
+    private final ProjectType projectType;
+    private final PrintStream logger;
+    private Build browserStackBuild;
+
+    public BrowserStackReportForBuild(final Run<?, ?> build, final ProjectType projectType, final String buildName, final PrintStream logger) {
+        super();
+        setBuild(build);
+        this.buildName = buildName;
+        this.browserStackSessions = new ArrayList<>();
+        this.result = new ArrayList<>();
+        this.resultMeta = new ArrayList<>();
+        this.resultAggregation = new JSONObject();
+        this.projectType = projectType;
+        this.logger = logger;
+        fetchBuildAndSessions();
+    }
+
+    private void fetchBuildAndSessions() {
+        final BrowserStackBuildAction browserStackBuildAction = getBuild().getAction(BrowserStackBuildAction.class);
+        if (browserStackBuildAction == null) {
+            log(logger, "Error: No BrowserStackBuildAction found");
+            return;
+        }
+
+        final BrowserStackCredentials credentials = browserStackBuildAction.getBrowserStackCredentials();
+        if (credentials == null) {
+            log(logger, "Error: BrowserStack credentials could not be fetched");
+            return;
+        }
+
+        BrowserStackClient client;
+        if (projectType == ProjectType.APP_AUTOMATE) {
+            client = new AppAutomateClient(credentials.getUsername(), credentials.getDecryptedAccesskey());
+        } else {
+            client = new AutomateClient(credentials.getUsername(), credentials.getDecryptedAccesskey());
+        }
+
+        this.browserStackBuild = fetchBrowserStackBuild(client, this.buildName);
+        if (this.browserStackBuild != null) {
+            this.browserStackSessions.addAll(fetchBrowserStackSessions(client, this.browserStackBuild.getId()));
+        }
```

Comment:
```suggestion
        Optional.ofNullable(browserStackBuild)
                .ifPresent(browserStackBuild -> 
                        browserStackSessions.addAll(fetchBrowserStackSessions(client, browserStackBuild.getId()))
                );
```

You can avoid checking null via branching instead prefer using power of `Optional`

Reply:
done